src/docs/perf/rendering/shader.md

```diff
@@ -0,0 +1,171 @@
+```
+
+iOS:
+```sh
+flutter build ios --bundle-sksl-path flutter_01.sksl.json
+```
+</li>
+
+<li markdown="1"> Test the newly built app.
+</li>
+</ol>
+
+Alternatively, you can write some integration tests to
+automate the first three steps using a single command.
+For example:
+
+```sh
+flutter drive --profile --cache-sksl --write-sksl-on-exit flutter_01.sksl.json -t test_driver/app.dart
+```
+
+With such [integration tests][], you can easily and reliably get the
+new SkSLs when the app code changes, or when Flutter upgrades.
+Such tests can also be used to verify the performance change
+before and after the SkSL warm-up. Even better, you can put
+those tests into a CI (continuous integration) system so the
+SkSLs are generated and tested automatically over the lifetime of an app.
+
+Take [Flutter Gallery][] as an example. The CI system is set up to
+generate SkSLs for every Flutter commit, and verifies the performance,
+in the [`transitions_perf_test.dart`][] test.
+For more details, see the [`flutter_gallery_sksl_warmup__transition_perf`][]
+and [`flutter_gallery_sksl_warmup_ios32__transition_perf`][] tasks.
+
+
+[Flutter Gallery]: {{site.github}}/flutter/flutter/tree/master/dev/integration_tests/flutter_gallery
+[`flutter_gallery_sksl_warmup__transition_perf`]: {{site.github}}/flutter/flutter/blob/master/dev/devicelab/bin/tasks/flutter_gallery_sksl_warmup__transition_perf.dart
+[`flutter_gallery_sksl_warmup_ios32__transition_perf`]: {{site.github}}/flutter/blob/master/dev/devicelab/bin/tasks/flutter_gallery_sksl_warmup_ios32__transition_perf.dart
+[integration tests]: /docs/cookbook/testing/integration/introduction
+[`transitions_perf_test.dart`]: {{site.github}}/flutter/flutter/blob/master/dev/integration_tests/flutter_gallery/test_driver/transitions_perf_test.dart
+
+## Limitations and considerations
+
+1. **Why not just compile or warm up _all_ possible shaders?**<br>
+   If there are only a limited number of possible shaders,
+   then Flutter could do a warm-up and compile all of them
+   before-hand to avoid such jank.
+   However, for the best overall performance,
+   the Skia GPU backend used by Flutter can dynamically generate
+   shaders based on a lot of parameters at runtime
+   (for example, draws, device models, and driver versions).
+   Due to all possible combinations of those parameters,
+   the number of possible shaders is exponential.
+   In short, Flutter uses programs (app, Flutter, and Skia code)
+   to generate some other programs (shaders). The number of possible
+   shader programs that Flutter can generate is simply too large.
+
+2. **Can SkSLs captured from one device help shader compilation jank
+   on another device?**<br>
+   Theoretically, there's no guarantee that the SkSLs from one device
+   would help on another device (but they also won't cause any troubles
+   if SkSLs aren't compatible across devices).
+   Practically, as shown in the table on this [SkSL-based warmup PR][],
```

Comment:
`[SkSL-based warmup PR][]` -> `[SkSL-based warmup issue][]`
(And update the link name below accordingly.)

Reply:
Done.